[Swift in WebKit] Use Swift for more _WKTextExtraction related types

rr-codes opened the pull request:
#### 681d4396dc308a5da4e7b48d4ca22a4903883d1a
<pre>
[Swift in WebKit] Use Swift for more _WKTextExtraction related types
<a href="https://bugs.webkit.org/show_bug.cgi?id=302999">https://bugs.webkit.org/show_bug.cgi?id=302999</a>
<a href="https://rdar.apple.com/165255871">rdar://165255871</a>

Reviewed by Wenson Hsieh.

Some _WKTextExtraction types are implemented in Swift, and others in Objective C. Bring consistency by implementing
them all in Swift.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtraction.mm: Removed.
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtraction.swift:
(_WKTextExtractionInteraction.nodeIdentifier):
(_WKTextExtractionInteraction.text):
(_WKTextExtractionInteraction.replaceAll):
(_WKTextExtractionInteraction.scrollToVisible):
(_WKTextExtractionInteraction.scrollDelta):
(_WKTextExtractionInteraction.backingLocation):
(_WKTextExtractionInteraction.location):
(_WKTextExtractionInteraction.hasSetLocation):
(_WKTextExtractionInteraction.debugDescription(in:)):
(_WKTextExtractionConfiguration.visibleTextOnly):
(_WKTextExtractionConfiguration.outputFormat):
(_WKTextExtractionConfiguration.nodeIdentifierInclusion):
(_WKTextExtractionConfiguration.filterOptions):
(_WKTextExtractionConfiguration.targetNode):
(_WKTextExtractionConfiguration.targetRect):
(_WKTextExtractionConfiguration.includeURLs):
(_WKTextExtractionConfiguration.includeRects):
(_WKTextExtractionConfiguration.includeEventListeners):
(_WKTextExtractionConfiguration.includeAccessibilityAttributes):
(_WKTextExtractionConfiguration.includeTextInAutoFilledControls):
(_WKTextExtractionConfiguration.mergeParagraphs):
(_WKTextExtractionConfiguration.skipNearlyTransparentContent):
(_WKTextExtractionConfiguration.onlyIncludeVisibleText):
(_WKTextExtractionConfiguration.maxWordsPerParagraph):
(_WKTextExtractionConfiguration.replacementStrings):
(_WKTextExtractionConfiguration.clientNodeAttributes):
(_WKTextExtractionConfiguration.forEachClientNodeAttribute(_:String:_WKJSHandle:)):
(_WKTextExtractionConfiguration.addClientAttribute(_:value:forNode:)):
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtractionInternal.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/303555@main">https://commits.webkit.org/303555@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ade16f9441ea3b0de655ade63943b48cdde8c06

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132505 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5000 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43560 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140024 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84490 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2c88e208-7e4d-4879-b947-90f021135b66) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134375 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5198 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4759 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101302 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68572 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f0d79b4a-9c99-4500-a2d6-fc339fb5fd04) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3664 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118697 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82096 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/37b0d111-8bad-438e-9423-6df8de9ad073) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3550 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1288 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83258 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/124575 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112453 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36813 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142676 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/131013 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4670 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37402 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109677 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4752 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4035 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109859 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27907 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3547 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114970 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58046 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4724 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33320 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163980 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4558 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68175 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/42596 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4815 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4681 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/API/Cocoa/_WKTextExtractionInternal.h:
```diff
@@ -25,12 +25,14 @@
 NS_ASSUME_NONNULL_BEGIN
 
+// This is equivalent to USE(APPLE_INTERNAL_SDK) || (!PLATFORM(WATCH) && !PLATFORM(APPLETV)
+#if (defined __has_include && __has_include(<CoreFoundation/CFPriv.h>)) || (!TARGET_OS_WATCH && !TARGET_OS_TV)
```
Hm... is this correct? (I would've expected `#if (defined(__has_include) && …`)

rr-codes replied:
I think so; it’s consistent with the definition in PlatformUse.h

```
#if PLATFORM(COCOA)
#if defined __has_include && __has_include(<CoreFoundation/CFPriv.h>)
#define USE_APPLE_INTERNAL_SDK 1
#endif
#endif
```

rr-codes committed:
[Swift in WebKit] Use Swift for more _WKTextExtraction related types
https://bugs.webkit.org/show_bug.cgi?id=302999
rdar://165255871

Reviewed by Wenson Hsieh.

Some _WKTextExtraction types are implemented in Swift, and others in Objective C. Bring consistency by implementing
them all in Swift.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtraction.mm: Removed.
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtraction.swift:
(_WKTextExtractionInteraction.nodeIdentifier):
(_WKTextExtractionInteraction.text):
(_WKTextExtractionInteraction.replaceAll):
(_WKTextExtractionInteraction.scrollToVisible):
(_WKTextExtractionInteraction.scrollDelta):
(_WKTextExtractionInteraction.backingLocation):
(_WKTextExtractionInteraction.location):
(_WKTextExtractionInteraction.hasSetLocation):
(_WKTextExtractionInteraction.debugDescription(in:)):
(_WKTextExtractionConfiguration.visibleTextOnly):
(_WKTextExtractionConfiguration.outputFormat):
(_WKTextExtractionConfiguration.nodeIdentifierInclusion):
(_WKTextExtractionConfiguration.filterOptions):
(_WKTextExtractionConfiguration.targetNode):
(_WKTextExtractionConfiguration.targetRect):
(_WKTextExtractionConfiguration.includeURLs):
(_WKTextExtractionConfiguration.includeRects):
(_WKTextExtractionConfiguration.includeEventListeners):
(_WKTextExtractionConfiguration.includeAccessibilityAttributes):
(_WKTextExtractionConfiguration.includeTextInAutoFilledControls):
(_WKTextExtractionConfiguration.mergeParagraphs):
(_WKTextExtractionConfiguration.skipNearlyTransparentContent):
(_WKTextExtractionConfiguration.onlyIncludeVisibleText):
(_WKTextExtractionConfiguration.maxWordsPerParagraph):
(_WKTextExtractionConfiguration.replacementStrings):
(_WKTextExtractionConfiguration.clientNodeAttributes):
(_WKTextExtractionConfiguration.forEachClientNodeAttribute(_:String:_WKJSHandle:)):
(_WKTextExtractionConfiguration.addClientAttribute(_:value:forNode:)):
* Source/WebKit/UIProcess/API/Cocoa/_WKTextExtractionInternal.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/303555@main
